### PR TITLE
feat(lint-python): use ruff

### PR DESCRIPTION
**Pull request by catcombo**
Relates to https://github.com/moneymeets/moneymeets-backoffice/pull/148
Lets keep it open until we finish testing of the `ruff` in the `backoffice-app`.

### CODE REVIEW

**Review by zyv — COMMENTED** [review]
What makes me a bit hesitant about this is what happens if this guy stops taking his pills. It doesn't seem to be an official action for ruff. But then again, maybe we should just accept it - the other way would be to fork it... what can possibly go wrong (tm) ?

**Comment by catcombo** [comment]
> What makes me a bit hesitant about this is what happens if this guy stops taking his pills. It doesn't seem to be an official action for ruff. But then again, maybe we should just accept it - the other way would be to fork it... what can possibly go wrong (tm) ?

I took it from the ruff documentation https://docs.astral.sh/ruff/usage/#github-action Of course, ruff developers could move it to the ruff organization to make it really official and under their control. Maybe they will do it later.

**Comment by zyv** [comment]
> > What makes me a bit hesitant about this is what happens if this guy stops taking his pills. It doesn't seem to be an official action for ruff. But then again, maybe we should just accept it - the other way would be to fork it... what can possibly go wrong (tm) ?
> 
> I took it from the ruff documentation https://docs.astral.sh/ruff/usage/#github-action Of course, ruff developers could move it to the ruff organization to make it really official and under their control. Maybe they will do it later.

Maybe you could be so kind to open an issue there to propose that?

**Comment by marns93** [comment]
@catcombo As discussed, we don't need to GitHub action `chartboost/ruff-action`, because it is just a wrapper to execute `ruff`. We are using our composite action anyway, so I would prefer to specify the inputs and just run the command without using the unofficial action.

**Comment by marns93** [comment]
@catcombo I've updated the action as suggested without using `chartboost/ruff-action`. Furthermore, I made it backwards compatible, so that we can merge this PR without breaking all repositories. 

Tested:

With flake8:
https://github.com/moneymeets/moneymeets-pulumi/actions/runs/6584963350/job/17890482382

With ruff:
https://github.com/moneymeets/moneymeets-backoffice/actions/runs/6584970332/job/17890503874

@felix11h FYI @IamAbbey @haidarCheaito

File: lint-python/action.yml [hunk 1]
```diff
@@ -1,20 +1,41 @@
 name: Lint Python
 description: Linting for Python projects
+
 inputs:
   exclude_trailing_comma_path:
     required: false
     description: Ignore trailing comma changes in given path
     default: ''
+  src:
```
Comment (zyv) [inline lint-python/action.yml]: I would call this `working_directory` or something like this.

**Review by felix11h — APPROVED** [review]
Approved from my side, we have tested this in monymeets-pulumi and moneymeets-backoffice. I think you can also convert the pull request from draft, @marns93. 

EDIT: Well, it was @catcombo's pull request initially anyway, so I just converted the pull request from draft myself :)

File: lint-python/action.yml [hunk 2]
```diff
@@ -1,20 +1,41 @@
...
+    description: Source to run ruff
+    default: '.'
 
 runs:
   using: composite
   steps:
-    - run: find ${{ inputs.exclude_trailing_comma_path }} -name '*.py' | xargs poetry run add-trailing-comma --exit-zero-even-if-changed --py36-plus
-      if: ${{ inputs.exclude_trailing_comma_path != '' }}
-      shell: bash
+      - run: echo "ruff-available=$(grep ruff pyproject.toml | wc -l)" >> $GITHUB_OUTPUT
```
Comment (catcombo) [inline lint-python/action.yml]: We can keep it like this if it's a temporary solution. But there is another way to check if package is available: `poetry show ruff`. The command returns non-zero status code if the package is not available.

File: lint-python/action.yml (hunk 1 above)
Comment (marns93) [reply lint-python/action.yml]: Updated to `working_directory` and the default value to `${{ github.workspace }}`.

File: lint-python/action.yml (hunk 2 above)
Comment (marns93) [reply lint-python/action.yml]: Good idea, changed it to `poetry show ruff -q ; echo $?`

File: lint-python/action.yml [hunk 3]
```diff
@@ -14,17 +18,33 @@ runs:
 
     - uses: moneymeets/moneymeets-composite-actions/check-git-diff@master
 
+    - run: echo "ruff-available=$(poetry show ruff -q ; echo $?)" >> $GITHUB_OUTPUT
+      id: detect-ruff
+      shell: bash
+
+    - name: Run linter
+      if: steps.detect-ruff.outputs.ruff-available == 0
```
Comment (zyv) [inline lint-python/action.yml:26]: Just wanted to say that it must be exactly the other way around, but you changed it already:

```
(moneymeets-backoffice-py3.11) zaytsev@parallels:~/src/moneymeets-backoffice$ poetry show ruff -q ; echo $?
0

(fondsnet-sdk-py3.11) zaytsev@parallels:~/src/fondsnet-sdk$ poetry show ruff -q ; echo $?
1
```

**Review by marns93 — APPROVED** [review]

File: lint-python/action.yml (hunk 3 above)
Comment (marns93) [reply lint-python/action.yml:26]: Yeah, already updated and tested. :)

**Review by zyv — APPROVED** [review]

**Pull request merged**